### PR TITLE
Enrich 4.1.0

### PR DESCRIPTION
**Pull request by spenes**
This PR makes the necessary changes and additions on docs for Enrich 4.1.0

### CODE REVIEW

File: docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Cross Navigation Enrichment"
+sidebar_position: 15
+sidebar_label: Cross Navigation
+---
+
+This enrichment parses the extended cross navigation format in `_sp` querystring parameter and attach the cross_navigation context to an event.
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md:7]:
```suggestion
This enrichment parses the extended cross navigation format in `_sp` querystring parameter and attaches the `cross_navigation` context to an event.
```

File: docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+title: "Cross Navigation Enrichment"
+sidebar_position: 15
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md:3]: Does this match the order in which it is run?

File: docs/enriching-your-data/available-enrichments/custom-javascript-enrichment/writing/index.md [hunk 3]
```diff
@@ -19,7 +19,9 @@ function process(event) {
 }
 ```
 
-Remember that you can only have _one_ JavaScript enrichment, and hence a single `process` function for your pipeline. However, you can split more complex logic into multiple helper functions and variables as you see fit, as long as you comply with the above interface.
+Starting with Enrich 4.1.0, it is possible to have multiple JavaScript enrichments.
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/custom-javascript-enrichment/writing/index.md:22]:
```suggestion
Starting with Enrich 4.1.0, it is possible to have multiple JavaScript enrichments. (Currently, the order in which they would run is not defined.)
```

File: docs/enriching-your-data/available-enrichments/custom-javascript-enrichment/writing/index.md [hunk 4]
```diff
@@ -277,3 +279,40 @@ function process(event) {
...
+```
+
+The parameter object can be accessed in JavaScript enrichment code via the second parameter of the `process` function, for example:
+
+```js
+function process(event, params) {
+  event.setApp_id(params.nested.bar);
+  return [];
+}
+```
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/custom-javascript-enrichment/writing/index.md:318]:
````suggestion
```
:::tip

This is useful when you want to quickly reconfigure the enrichment without updating the JavaScript code.

:::
````

File: docs/enriching-your-data/available-enrichments/index.md [hunk 5]
```diff
@@ -33,6 +33,7 @@ Currently, it’s not possible to configure more than one instance of the same e
...
 | [IP Anonymization](/docs/enriching-your-data/available-enrichments/ip-anonymization-enrichment/index.md)           | Anonymize the IP addresses found in the `user_ipaddress` field by replacing a certain number of octets with `x`s.                                                                                                              |
 | [PII Pseudonymization](/docs/enriching-your-data/available-enrichments/pii-pseudonymization-enrichment/index.md)   | Better protect the privacy rights of data subjects by psuedoanonymizing collected data.                                                                                                                                        |
+| [Cross Navigation Enrichment](/docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md)   | Parse the extended cross navigation format in _sp querystring parameter and attach the cross_navigation context to an event.                                                                                                                                        |
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/index.md:36]: Does this match the running order?

**Review by stanch — CHANGES_REQUESTED** [review]

File: docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+sidebar_label: Cross Navigation
+---
+
+This enrichment parses the extended cross navigation format in `_sp` querystring parameter and attach the cross_navigation context to an event.
+
+The extended cross navigation format can be described by `_sp={domainUserId}.{timestamp}.{sessionId}.{subjectUserId}.{sourceId}.{platform}.{reason}`
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md:9]: @adatzer could you suggest a few more sentences we can say about this to explain why the enrichment is needed to a general audience?

File: docs/enriching-your-data/available-enrichments/index.md (hunk 5 above)
Comment (spenes) [reply docs/enriching-your-data/available-enrichments/index.md]: Fixed this as well 👍

File: docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md (hunk 6 above)
Comment (spenes) [reply docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md:11]: I've added [Matus's suggestion](https://github.com/snowplow/engineering-announcements/pull/106#discussion_r1494746906). Let me know if you want to add anything other.

File: docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Cross Navigation Enrichment"
+sidebar_position: 5
+sidebar_label: Cross Navigation
+---
+
+This enrichment parses the extended cross navigation format in `_sp` querystring parameter and attaches the `cross_navigation` context to an event.
+
+The `_sp` parameter can be attached by our Web ([see cross-domain tracking](https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/cross-domain-tracking/)) and [mobile trackers](https://docs.snowplow.io/docs/collecting-data/collecting-from-own-applications/mobile-trackers/tracking-events/session-tracking/#decorating-outgoing-links-using-cross-navigation-tracking) and contains user, session and app identifiers (e.g., domain user and session IDs, business user ID, source app ID). The information to include in the parameters is configurable in the trackers. This is useful for tracking the movement of users across different apps and platforms.
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md:9]:
```suggestion
The `_sp` parameter can be attached by our Web ([see cross-domain tracking](/docs/collecting-data/collecting-from-own-applications/javascript-trackers/web-tracker/cross-domain-tracking/index.md)) and [mobile trackers](/docs/collecting-data/collecting-from-own-applications/mobile-trackers/tracking-events/session-tracking/index.md#decorating-outgoing-links-using-cross-navigation-tracking) and contains user, session and app identifiers (e.g., domain user and session IDs, business user ID, source app ID). The information to include in the parameters is configurable in the trackers. This is useful for tracking the movement of users across different apps and platforms.
```

File: docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+This enrichment extracts `_sp` querystring parameter from the following inputs:
+
+- The `page_url` field from the Snowplow event
+- The referer uri extracted from corresponding HTTP header in the raw event
+
+## Output
+
+This enrichment adds a new derived context to the enriched event with [this schema](https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow/cross_navigation/jsonschema/1-0-0).
+
+Also, it continues to populate `refr_domain_userid` and `refr_tstamp` enriched event fields as before.
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md:37]: What does “as before” mean here? Compared to what? Maybe we just say it populates those fields (without “continues” and “as before”)?

File: docs/enriching-your-data/available-enrichments/index.md [hunk 9]
```diff
@@ -23,6 +23,7 @@ Currently, it’s not possible to configure more than one instance of the same e
...
 | [Referer Parser](/docs/enriching-your-data/available-enrichments/referrer-parser-enrichment/index.md)              | Extracts attribution data from referer URLs.                                                                                                                                                                                   |
 | [Campaign Attribution](/docs/enriching-your-data/available-enrichments/campaign-attribution-enrichment/index.md)   | Choose which querystring parameters will be used to generate the marketing campaign fields. *If you do not enable the campaign_attribution enrichment, those fields will not be populated.*                                    |
+| [Cross Navigation Enrichment](/docs/enriching-your-data/available-enrichments/cross-navigation-enrichment/index.md)   | Parse the extended cross navigation format in _sp querystring parameter and attach the cross_navigation context to an event.                                                                                                                                        |
```
Comment (stanch) [inline docs/enriching-your-data/available-enrichments/index.md:26]: Could you state here that it’s new as of Enrich 4.1.0? Also, could you update `docs/reusable/enrichment-list/_index.md` the same way?

**Review by stanch — CHANGES_REQUESTED** [review]

File: docs/enriching-your-data/available-enrichments/index.md (hunk 9 above)
Comment (spenes) [reply docs/enriching-your-data/available-enrichments/index.md]: I've updated `docs/reusable/enrichment-list/_index.md` but couldn't find any place it is used. Should we remove it ?

Comment (stanch) [reply docs/enriching-your-data/available-enrichments/index.md]: Oh! Yes, then we should remove, sorry!

**Review by stanch — APPROVED** [review]

**Review by stanch — APPROVED** [review]

**Pull request merged**